### PR TITLE
Tests: Tests for MakeBucketRequest

### PR DESCRIPTION
**Pull request by hackintoshrao**
- Test for makeBucketRequest
- Test for `MakeBucket` is covered in the functional tests, since `makeBucketRequest` forms the core of `MakeBucket` there is no need for unit test for it. Since tests running on S3 is time consuming its better to just keep them in functional tests.

### CODE REVIEW

File: api-put-bucket_test.go [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+	"encoding/base64"
+	"encoding/hex"
+	"encoding/xml"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"testing"
+)
+
+func createExpectedRequest(c *Client, bucketName string, location string, req *http.Request) (*http.Request, error) {
```
Comment (harshavardhana) [inline api-put-bucket_test.go]: Add a comment here.

File: api-put-bucket_test.go [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+		{"localhost:9000", "my-access-key", "my-secret-key", false},
+		// one of acess-key and secret-key are empty.
+		{"localhost:9000", "", "my-secret-key", false},
+
+		// endpoint amazon s3.
+		{"s3.amazonaws.com", "", "", false},
+		{"s3.amazonaws.com", "my-access-key", "my-secret-key", false},
+		{"s3.amazonaws.com", "my-acess-key", "", false},
+
+		// endpoint google bucket storage.
```
Comment (harshavardhana) [inline api-put-bucket_test.go]: google cloud storage not 'bucket storage'

**Comment by harshavardhana** [comment]
Everything else looks good. Just those two comments.

**Comment by hackintoshrao** [comment]
Will push in a minute.

**Pull request merged**